### PR TITLE
6pm 3 quarter filter- in quarter filter page, can only search courses using correct quarter format (e.g 'W21') 

### LINKED ISSUE #121: 'Set Quarter Filter' Page should not take in numeric quarter format
# User Story

As a admin I can search courses by quarter so that I can see course history.

# Acceptance Criteria

- [x] Search can be done using common format and not numeric format (see note 1)
- [x] On the search filter page, an admin may ONLY enter a quarter in Qyy format (e.g. W21, S21).  No other format is accepted.  As an alternative, use a quarter selector that selects from among the quarters that are in the database (there an api endpoint to provide those), make it a drop down item selector.

Note 1: Right now you can filter but you have to type the quarter in numeric format (i.e 20211 for Winter 2021) but you should only allow searches that use the common format - W21 or S21, etc.

# Implementation Todos

Front end:

- [x] Change the placeholder text to be more descriptive by adding example search
- [x] Show field validation for search - must be a valid quarter (Look at Add Courses page, you can probably reuse)
- [x] Make sure "Currently Filtering by quarter ___" shows correctly after previous changes. Ex: "Currently filtering by quarter W21" and not of "Currently filtering by quarter 20211"

Back end:

- [x] Convert text for search to numeric format for endpoint call
- [x] Add field validation for search

Testing:
- [x] Backend search tests still work as expected
- [x] Error tests if user puts in wrong search input type
- [x] All previous tests pass

### PR DESCRIPTION
**Pull request by pranav01acharya**
Fixes #121  . Now, admin can search courses by quarter in the quarter filter page using only the correct format (eg 'W21') and not any other format (eg '20211'). Screenshots below:
Screen Shot 2021-03-09 at 7 01 45 PM
Screen Shot 2021-03-09 at 7 01 52 PM
Screen Shot 2021-03-09 at 7 02 17 PM
Screen Shot 2021-03-09 at 7 02 29 PM

### CODE REVIEW

**Review by Bryanz2019 — DISMISSED** [review]
LGTM

**Review by Bryanz2019 — APPROVED** [review]
LGTM

File: javascript/src/test/components/QuarterFilterForm.test.js [hunk 1]
```diff
@@ -28,7 +32,71 @@ describe("QuarterFilterForm tests", () => {
     userEvent.click(submitButton);
 
     expect(upsertFilterMock).toHaveBeenCalledTimes(1);
-    expect(upsertFilterMock).toHaveBeenCalledWith(sampleFilterVal);
   });
+  test("set quarter filter format works with right format", async () => {
```
Comment (JacquelineMai) [inline javascript/src/test/components/QuarterFilterForm.test.js:36]: For these 4 tests, verify that there is actual corresponding correct behavior, not just that the filter is called. For example, for the 3 tests with invalid values, verify that the error message is actually displayed on the page after submission. For the correct test, verify that the error message is not present after submission

**Review by JacquelineMai — CHANGES_REQUESTED** [review]

**Review by Bryanz2019 — APPROVED** [review]
LGTM

**Review by JacquelineMai — APPROVED** [review]
nicee

**Pull request merged**